File: src/mffs/item/module/interdiction/ItemModuleWarn.java
package mffs.item.module.interdiction;

import mffs.api.security.IInterdictionMatrix;
import net.minecraft.entity.EntityLiving;
import net.minecraft.entity.player.EntityPlayer;

public class ItemModuleWarn extends ItemModuleInterdictionMatrix
{
	public ItemModuleWarn(int i)
	{
		super(i, "moduleWarn");
	}

	@Override
	public boolean onDefend(IInterdictionMatrix interdictionMatrix, EntityLiving entityLiving)
	{
		boolean hasPermission = false;
		if (!hasPermission && entityLiving instanceof EntityPlayer)
		{
			((EntityPlayer) entityLiving).addChatMessage("[" + interdictionMatrix.getInvName() + "] Leave this zone immediately. You have no right to enter.");
		}

		return false;
	}
}

File: src/main/scala/mffs/security/module/ItemModuleWarn.scala
package mffs.security.module

import net.minecraft.entity.EntityLivingBase
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.util.ChatComponentText
import resonant.api.mffs.security.IInterdictionMatrix
import resonant.lib.utility.LanguageUtility

class ItemModuleWarn extends ItemModuleDefense
{
  override def onDefend(interdictionMatrix: IInterdictionMatrix, entityLiving: EntityLivingBase): Boolean =
  {
    val hasPermission: Boolean = false

    if (!hasPermission && entityLiving.isInstanceOf[EntityPlayer])
    {
      (entityLiving.asInstanceOf[EntityPlayer]).addChatMessage(new ChatComponentText("[" + interdictionMatrix.getInventoryName + "] " + LanguageUtility.getLocal("message.moduleWarn.warn")))
    }

    return false
  }
}